<!DOCTYPE html>
<html lang="en">

  <!-- Head -->
  <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">    <!-- Metadata, OpenGraph and Schema.org -->
    

    <!-- Standard metadata -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Publications | Center for Human Brain Variation</title>
    <meta name="author" content="Center for Human Brain Variation">
    <meta name="description" content="Experimental and computational methods that make the science at the Center possible.">


    <!-- Bootstrap & MDB -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous">

    <!-- Fonts & Icons -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.4.2/css/all.min.css" integrity="sha256-CTSx/A06dm1B063156EVh15m6Y67pAjZZaQc89LLSrU=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

    <!-- Code Syntax Highlighting -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light">

    <!-- Styles -->
    
    <link rel="stylesheet" href="/assets/css/main.css?v=1731475699">

    <link rel="canonical" href="https://brain.broadinstitute.org/publications/">
    
    <!-- Dark Mode -->
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark">

    <script src="/assets/js/theme.js"></script>
    <script src="/assets/js/dark_mode.js"></script>
    

    <!-- favicons -->
    <link rel="apple-touch-icon" sizes="180x180" href="/assets/img/apple-touch-icon.png">
    <link rel="icon" type="image/png" sizes="32x32" href="/assets/img/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/assets/img/favicon-16x16.png">
    <link rel="manifest" href="/assets/img/site.webmanifest">
    <link rel="mask-icon" href="/assets/img/safari-pinned-tab.svg" color="#5bbad5">
    <link rel="shortcut icon" href="/assets/img/favicon.ico">
    <meta name="msapplication-TileColor" content="#da532c">
    <meta name="msapplication-config" content="/assets/img/browserconfig.xml">
    <meta name="theme-color" content="#ffffff">

  </head>

  <!-- Body -->
  <body class="fixed-top-nav sticky-bottom-footer d-flex flex-column min-vh-100">

    <!-- Header -->    <header>
      <!-- Banner -->
      <div id="site-banner" class="banner" style="display: none">
          <div class="banner_content">
              <div class="banner_text">
                  The next CHBV seminar is on Friday, December 6.  Register and learn more <a href="/seminar-series/">here</a>!
              </div>
              <button id="site-banner-close" class="banner_close" type="button">
                  <i class="fa fa-window-close" aria-hidden="true"></i>
              </button>
          </div>
      </div>
      <script>
      if (sessionStorage.getItem('bannerClosed') !== 'true') {
          document.getElementById('site-banner').style.display = 'block';
      }
      document.addEventListener('DOMContentLoaded', function () {
          const banner = document.getElementById('site-banner');
          const closeButton = document.getElementById('site-banner-close');
          // Add event listener to close button
          closeButton.addEventListener('click', function() {
              banner.style.display = 'none';
              // Store the state in sessionStorage
              sessionStorage.setItem('bannerClosed', 'true');
          });
      });
      </script>
      <!-- Nav Bar -->
      <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
        <div class="container">
          <a href="/"><img src="/assets/img/logo_light.png" alt="Center for Human Brain Variation logo" height="40px" id="logo"></a>
          <!-- Navbar Toggle -->
          <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar top-bar"></span>
            <span class="icon-bar middle-bar"></span>
            <span class="icon-bar bottom-bar"></span>
          </button>

          <div class="collapse navbar-collapse text-right" id="navbarNav">
            <ul class="navbar-nav ml-auto flex-nowrap">

              <!-- Other pages -->
              <li class="nav-item ">
                <a class="nav-link" href="/about/">About</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/team/">Team</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/research/">Research</a>
              </li>
              <li class="nav-item active">
                <a class="nav-link" href="/publications/">Publications<span class="sr-only">(current)</span></a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/news/">News</a>
              </li>
              <li class="nav-item dropdown ">
                <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Outreach</a>
                <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
                  <a class="dropdown-item" href="/outreach/">Outreach and education</a>
                  <a class="dropdown-item" href="/seminar-series/">Seminar Series</a>
                </div>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/join-us/">Join</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/contact/">Contact</a>
              </li>
            </ul>
          </div>
        </div>
      </nav>


      <!-- Scrolling Progress Bar -->
      <progress id="progress" value="0">
        <div class="progress-container">
          <span class="progress-bar"></span>
        </div>
      </progress>
    </header>


    <!-- Content -->
    <main class="flex-fill container mt-5">
      <!-- page.html -->
        <div class="post">

          <header class="post-header">
            <h1 class="post-title">Publications</h1>
            <p class="post-description">Experimental and computational methods that make the science at the Center possible.</p>
          </header>

          <article class="my_contact">
            <!-- _pages/publications.md -->
<div class="publications">
  <h2 class="year">2023</h2>
  <ol class="bibliography">
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/fleming_et_al.nature_methods.2023-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/fleming_et_al.nature_methods.2023-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/fleming_et_al.nature_methods.2023-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/publication_preview/fleming_et_al.nature_methods.2023.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="fleming_et_al.nature_methods.2023.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="Fleming2023" class="col-sm-8">
        <!-- Title -->
        <div class="title"><a href="https://doi.org/10.1038/s41592-023-01943-7" target="_blank" rel="noopener">Unsupervised removal of systematic background noise from droplet-based single-cell experiments using CellBender</a></div>
        <!-- Author -->
        <div class="author">
        

        Stephen J. Fleming, Mark D. Chaffin, Alessandro Arduini, and
          <span class="more-authors" title="click to view 6 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '6 more authors' ? 'Amer-Denis Akkad, Eric Banks, John C. Marioni, Anthony A. Philippakis, Patrick T. Ellinor, Mehrtash Babadi' : '6 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text);
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '');
              ">6 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Nature Methods</em>, 2023
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/assets/pdf/fleming_et_al.nature_methods.2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.1038/s41592-023-01943-7"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.1038/s41592-023-01943-7" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Droplet-based single-cell assays, including single-cell RNA sequencing (scRNA-seq), single-nucleus RNA sequencing (snRNA-seq) and cellular indexing of transcriptomes and epitopes by sequencing (CITE-seq), generate considerable background noise counts, the hallmark of which is nonzero counts in cell-free droplets and off-target gene expression in unexpected cell types. Such systematic background noise can lead to batch effects and spurious differential gene expression results. Here we develop a deep generative model based on the phenomenology of noise generation in droplet-based assays. The proposed model accurately distinguishes cell-containing droplets from cell-free droplets, learns the background noise profile and provides noise-free quantification in an end-to-end fashion. We implement this approach in the scalable and robust open-source software package CellBender. Analysis of simulated data demonstrates that CellBender operates near the theoretically optimal denoising limit. Extensive evaluations using real datasets and experimental benchmarks highlight enhanced concordance between droplet-based single-cell data and established gene expression patterns, while the learned background noise profile provides evidence of degraded or uncaptured cell types.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/wells_et_al.cell_stem_cell.2023-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/wells_et_al.cell_stem_cell.2023-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/wells_et_al.cell_stem_cell.2023-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/publication_preview/wells_et_al.cell_stem_cell.2023.jpg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="wells_et_al.cell_stem_cell.2023.jpg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="WELLS2023312" class="col-sm-8">
        <!-- Title -->
        <div class="title"><a href="https://www.sciencedirect.com/science/article/pii/S1934590923000358" target="_blank" rel="noopener">Natural variation in gene expression and viral susceptibility revealed by neural progenitor cell villages</a></div>
        <!-- Author -->
        <div class="author">
        

        Michael F. Wells, James Nemesh, Sulagna Ghosh, and
          <span class="more-authors" title="click to view 24 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '24 more authors' ? 'Jana M. Mitchell, Max R. Salick, Curtis J. Mello, Daniel Meyer, Olli Pietilainen, Federica Piccioni, Ellen J. Guss, Kavya Raghunathan, Matthew Tegtmeyer, Derek Hawes, Anna Neumann, Kathleen A. Worringer, Daniel Ho, Sravya Kommineni, Karrie Chan, Brant K. Peterson, Joseph J. Raymond, John T. Gold, Marco T. Siekmann, Emanuela Zuccaro, Ralda Nehme, Ajamete Kaykas, Kevin Eggan, Steven A. McCarroll' : '24 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text);
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '');
              ">24 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Cell Stem Cell</em>, 2023
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/assets/pdf/wells_et_al.cell_stem_cell.2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="https://doi.org/10.1016/j.stem.2023.01.010"></span>
              <span class="__dimensions_badge_embed__" data-doi="https://doi.org/10.1016/j.stem.2023.01.010" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Summary
  Human genome variation contributes to diversity in neurodevelopmental outcomes and vulnerabilities; recognizing the underlying molecular and cellular mechanisms will require scalable approaches. Here, we describe a “cell village” experimental platform we used to analyze genetic, molecular, and phenotypic heterogeneity across neural progenitor cells from 44 human donors cultured in a shared in vitro environment using algorithms (Dropulation and Census-seq) to assign cells and phenotypes to individual donors. Through rapid induction of human stem cell-derived neural progenitor cells, measurements of natural genetic variation, and CRISPR-Cas9 genetic perturbations, we identified a common variant that regulates antiviral IFITM3 expression and explains most inter-individual variation in susceptibility to the Zika virus. We also detected expression QTLs corresponding to GWAS loci for brain traits and discovered novel disease-relevant regulators of progenitor proliferation and differentiation such as CACHD1. This approach provides scalable ways to elucidate the effects of genes and genetic variation on cellular phenotypes.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/assets/img/publication_preview/russell_et_al.biorxiv.2023-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/assets/img/publication_preview/russell_et_al.biorxiv.2023-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/assets/img/publication_preview/russell_et_al.biorxiv.2023-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/assets/img/publication_preview/russell_et_al.biorxiv.2023.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="russell_et_al.biorxiv.2023.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="Russell2023.04.01.535228" class="col-sm-8">
        <!-- Title -->
        <div class="title"><a href="https://www.biorxiv.org/content/early/2023/04/03/2023.04.01.535228" target="_blank" rel="noopener">Slide-tags: scalable, single-nucleus barcoding for multi-modal spatial genomics</a></div>
        <!-- Author -->
        <div class="author">
        

        Andrew J. C. Russell, Jackson A. Weir, Naeem M. Nadaf, and
          <span class="more-authors" title="click to view 14 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '14 more authors' ? 'Matthew Shabet, Vipin Kumar, Sandeep Kambhampati, Ruth Raichur, Giovanni J. Marrero, Sophia Liu, Karol S. Balderrama, Charles R. Vanderburg, Vignesh Shanmugam, Luyi Tian, Catherine J. Wu, Charles H. Yoon, Evan Z. Macosko, Fei Chen' : '14 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text);
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '');
              ">14 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>bioRxiv</em>, 2023
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/assets/pdf/russell_et_al.biorxiv.2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-arxiv-id="https://www.biorxiv.org/content/early/2023/04/03/2023.04.01.535228.full.pdf"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.1101/2023.04.01.535228" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Recent technological innovations have enabled the high-throughput quantification of gene expression and epigenetic regulation within individual cells, transforming our understanding of how complex tissues are constructed. Missing from these measurements, however, is the ability to routinely and easily spatially localise these profiled cells. We developed a strategy, Slide-tags, in which single nuclei within an intact tissue section are ‘tagged’ with spatial barcode oligonucleotides derived from DNA-barcoded beads with known positions. These tagged nuclei can then be used as input into a wide variety of single-nucleus profiling assays. Application of Slide-tags to the mouse hippocampus positioned nuclei at less than 10 micron spatial resolution, and delivered whole-transcriptome data that was indistinguishable in quality from ordinary snRNA-seq. To demonstrate that Slide-tags can be applied to a wide variety of human tissues, we performed the assay on brain, tonsil, and melanoma. We revealed cell-type-specific spatially varying gene expression across cortical layers and spatially contextualised receptor-ligand interactions driving B-cell maturation in lymphoid tissue. A major benefit of Slide-tags is that it is easily adaptable to virtually any single-cell measurement technology. As proof of principle, we performed multiomic measurements of open chromatin, RNA, and T-cell receptor sequences in the same cells from metastatic melanoma. We identified spatially distinct tumour subpopulations to be differentially infiltrated by an expanded T-cell clone and undergoing cell state transition driven by spatially clustered accessible transcription factor motifs. Slide-tags offers a universal platform for importing the compendium of established single-cell measurements into the spatial genomics repertoire.Competing Interest StatementE.Z.M. and F.C. are academic founders of Curio Bioscience. F.C., E.Z.M., A.J.C.R., J.A.W., N.M.N., and V.K. are listed as inventors on a patent application related to the work.</p>
          </div>
        </div>
      </div>
</li>
</ol>


</div>

          </article>

        </div>

    </main>

    <!-- Footer -->    <footer class="sticky-bottom  mt-auto py-3">
      <div class="container">
            <div>Contact us</div>
            <div>
<a href="mailto:%62%72%61%69%6E@%62%72%6F%61%64%69%6E%73%74%69%74%75%74%65.%6F%72%67" title="email"><i class="fas fa-envelope"></i></a> <a href="mailto:%62%72%61%69%6E@%62%72%6F%61%64%69%6E%73%74%69%74%75%74%65.%6F%72%67" title="email">brain@broadinstitute.org</a>
</div>
            <div>
<a href="https://twitter.com/BroadBrainCtr" title="Twitter" rel="external nofollow noopener" target="_blank"><i class="fab fa-x-twitter"></i></a> <a href="https://twitter.com/BroadBrainCtr" title="Twitter" rel="external nofollow noopener" target="_blank">BroadBrainCtr</a>
</div>
            <div>
<a href="https://www.linkedin.com/company/BroadBrainCenter" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fab fa-linkedin"></i></a> <a href="https://www.linkedin.com/company/BroadBrainCenter" title="LinkedIn" rel="external nofollow noopener" target="_blank">BroadBrainCenter</a>
</div>

        <div class="copywrite">
          © Copyright 2024 Center for Human Brain Variation. 
        </div>
      </div>
    </footer>


    <!-- JavaScripts -->
    <!-- jQuery -->
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

    <!-- Bootsrap & MDB scripts -->
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script>

    <!-- Masonry & imagesLoaded -->
  <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
  <script defer src="/assets/js/masonry.js" type="text/javascript"></script>
    
  <!-- Medium Zoom JS -->
  <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script>
  <script defer src="/assets/js/zoom.js"></script><!-- Load Common JS -->
  <script defer src="/assets/js/common.js"></script>
  <script defer src="/assets/js/copy_code.js" type="text/javascript"></script>

    
  <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
  <script async src="https://badge.dimensions.ai/badge.js"></script>

    <!-- MathJax -->
  <script type="text/javascript">
    window.MathJax = {
      tex: {
        tags: 'ams'
      }
    };
  </script>
  <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script>

    
    

<!-- Scrolling Progress Bar -->
<script type="text/javascript">
  /*
   * This JavaScript code has been adapted from the article 
   * https://css-tricks.com/reading-position-indicator/ authored by Pankaj Parashar, 
   * published on the website https://css-tricks.com on the 7th of May, 2014.
   * Couple of changes were made to the original code to make it compatible 
   * with the `al-foio` theme.
   */
  const progressBar = $("#progress");
  /*
   * We set up the bar after all elements are done loading.
   * In some cases, if the images in the page are larger than the intended
   * size they'll have on the page, they'll be resized via CSS to accomodate
   * the desired size. This mistake, however, breaks the computations as the
   * scroll size is computed as soon as the elements finish loading.
   * To account for this, a minimal delay was introduced before computing the
   * values.
   */
  window.onload = function () {
    setTimeout(progressBarSetup, 50);
  };
  /*
   * We set up the bar according to the browser.
   * If the browser supports the progress element we use that.
   * Otherwise, we resize the bar thru CSS styling
   */
  function progressBarSetup() {
    if ("max" in document.createElement("progress")) {
      initializeProgressElement();
      $(document).on("scroll", function() {
        progressBar.attr({ value: getCurrentScrollPosition() });
      });
      $(window).on("resize", initializeProgressElement);
    } else {
      resizeProgressBar();
      $(document).on("scroll", resizeProgressBar);
      $(window).on("resize", resizeProgressBar);
    }
  }
  /*
   * The vertical scroll position is the same as the number of pixels that
   * are hidden from view above the scrollable area. Thus, a value > 0 is
   * how much the user has scrolled from the top
   */
  function getCurrentScrollPosition() {
    return $(window).scrollTop();
  }

  function initializeProgressElement() {
    let navbarHeight = $("#navbar").outerHeight(true);
    $("body").css({ "padding-top": navbarHeight });
    $("progress-container").css({ "padding-top": navbarHeight });
    progressBar.css({ top: navbarHeight });
    progressBar.attr({
      max: getDistanceToScroll(),
      value: getCurrentScrollPosition(),
    });
  }
  /*
   * The offset between the html document height and the browser viewport
   * height will be greater than zero if vertical scroll is possible.
   * This is the distance the user can scroll
   */
  function getDistanceToScroll() {
    return $(document).height() - $(window).height();
  }

  function resizeProgressBar() {
    progressBar.css({ width: getWidthPercentage() + "%" });
  }
  // The scroll ratio equals the percentage to resize the bar
  function getWidthPercentage() {
    return (getCurrentScrollPosition() / getDistanceToScroll()) * 100;
  }
</script>

  </body>
</html>
